library/eddsa.c

```diff
@@ -0,0 +1,590 @@
+/*
+ *  Edwards-curve Digital Signature Algorithm
+ *
+ *  Copyright The Mbed TLS Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * References:
+ *
+ * SEC1 http://www.secg.org/index.php?action=secg,docs_secg
+ */
+
+#include "common.h"
+
+#if defined(MBEDTLS_EDDSA_C)
+
+#include "mbedtls/eddsa.h"
+#include "mbedtls/asn1write.h"
+
+#include <string.h>
+
+#if defined(MBEDTLS_PLATFORM_C)
+#include "mbedtls/platform.h"
+#else
+#include <stdlib.h>
+#define mbedtls_calloc    calloc
+#define mbedtls_free       free
+#endif
```

Comment:
We've realized that the conditional inclusion of `platform.h` on `MBEDTLS_PLATFORM_C` was not a good idea, and [we're moving to unconditional inclusion](https://github.com/Mbed-TLS/mbedtls/pull/6291). Please use the new style in this new file.
```suggestion
#include "mbedtls/platform.h"
```